broker-plugins/jdbc-store/src/main/java/org/apache/qpid/server/store/jdbc/AbstractJDBCMessageStore.java

```diff
@@ -216,7 +216,7 @@ private void updateDbVersion(int newVersion) throws SQLException
     protected void initMessageStore(final ConfiguredObject<?> parent)
     {
         _parent = parent;
-        _executor = new ScheduledThreadPoolExecutor(4, new ThreadFactory()
+        _executor = new ScheduledThreadPoolExecutor(Runtime.getRuntime().availableProcessors(), new ThreadFactory()
```

Comment:
Olivier,
I am wandering whether it would be better to introduce a context variable for configuring the number of core threads in executor. Potentially, the default value can be set to 'availableProcessors' but I am a bit concerned about setting core threads number to 'availableProcessors'. On machines with hundreds of CPUs that might result in wasted system resources. Please note, that we are starting all core threads immediately after executor creation.  I think, that for this kind of tasks we need a special executor which can scale threads up and down when required.  Though, I do not want to spend time on such executor implementation.  

I created a pull request #32 where I moved setting of "qpid.jdbcstore.inClauseMaxSize" into context variables. I think we need to do the same for the number of core threads for the pool executor.  For example, we can introduce context variable like "qpid.jdbcstore.commitExecutorThreads". As we do not have executor capable of scaling number of threads up and down, I suppose, we can set the default value to Runtime.getRuntime().availableProcessors().

What do you think about suggested changes?

I am planning to commit PR #32 tonight if you have no objection

Reply:
I agree that hardcoding this value to 'availableProcessors' might not suit everyone.
I'll do the change next week when I come back.

Reply:
I used qpid.jdbcstore.executorThreads because this executor is not just used for the commit but also for the batched deletes